src/Daemon/SentryWriter.cpp

```diff
@@ -133,35 +141,43 @@ void SentryWriter::initialize(Poco::Util::LayeredConfiguration & config)
     }
 }
 
-void SentryWriter::shutdown()
+SentryWriter::~SentryWriter()
 {
     if (initialized)
         sentry_shutdown();
 }
 
-void SentryWriter::onFault(int sig, const std::string & error_message, const StackTrace & stack_trace)
+void SentryWriter::onFault(int sig_or_error, const std::string & error_message, const FramePointers & frame_pointers, size_t offset, size_t size)
 {
     auto logger = getLogger("SentryWriter");
     if (initialized)
     {
         sentry_value_t event = sentry_value_new_message_event(SENTRY_LEVEL_FATAL, "fault", error_message.c_str());
-        sentry_set_tag("signal", strsignal(sig)); // NOLINT(concurrency-mt-unsafe) // not thread-safe but ok in this context
-        sentry_set_extra("signal_number", sentry_value_new_int32(sig));
+        if (sig_or_error > 0)
+        {
+            int sig = sig_or_error;
+            sentry_set_tag("signal", strsignal(sig)); // NOLINT(concurrency-mt-unsafe) // not thread-safe but ok in this context
+            sentry_set_extra("signal_number", sentry_value_new_int32(sig));
+        }
+        else
+        {
+            /// Can be only LOGICAL_ERROR, but just in case.
+            int code = -sig_or_error;
+            sentry_set_tag("exception", DB::ErrorCodes::getName(code).data());
+            sentry_set_extra("exception_code", sentry_value_new_int32(code));
+        }
```

Comment:
I think it is better to separate not just by the sign of the things. There are two possibilities in my opinion:
1. keep thet `onFault` for signals and make another function for exceptions
2. introduce an enum/boolean flag to `onFault` that indicates whether it is signal or error

Reply:
Done.